test/trails_test2_handler.erl

```diff
@@ -3,10 +3,12 @@
 -behaviour(trails_handler).
 
 %% API
--export([trails/0]).
+-export([trails/0, trails/1]).
 
 trails() ->
   [
    {"/api/resource3/[:id]", trails_test2_handler, []},
    {"/api/:id/resource4", trails_test2_handler, [arg0]}
   ].
+
+trails(_Opts) -> [].
```

Comment:
Now… if you make this callback optional, you won't need to add this boilerplate here.